contrib/auto-render/auto-render.js

```diff
@@ -2,6 +2,8 @@
 /* global katex */
 
 const splitAtDelimiters = require("./splitAtDelimiters");
+const Settings = require("./../../src/Settings");
+const utils = require("./../../src/utils");
```

Comment:
Hmmm… I'd consider these files to be internal to KaTeX' codebase. Now we have a dependency here, which means that the contributed auto renderer may need to be adjusted when and if the files you reference get modified. It also means duplicating some code, as browserify would include those files into both its outputs. So I'd check very carefully whether that's worth the cost.

You're including `utils` for [`utils.dflt`](https://github.com/alexpearce/KaTeX/blob/43ac3d3987a7c8c73baa059303cd5a83655099a6/src/utils.js#L34-L39). The main benefit of that it that it falls back to the default whenever the argument is `undefined`. But the values expected here are arrays. Which means that a simple `||` might work just as well. If you consider `util.dflt` I would rather copy&paste that one function than having the inter-dependency between code bases.

I'd advise against a [`Settings`](https://github.com/alexpearce/KaTeX/blob/43ac3d3987a7c8c73baa059303cd5a83655099a6/src/Settings.js) object. That class is used internally, but the external interface is a simple object (oten a literal) called `options`. I see no benefit in using a `Settings` object in the public interface, and a ton of problems if we ever decide to add stuff to `Settings`.

Just my 2¢.